server/leafnode.go

```diff
@@ -1670,7 +1670,8 @@ func (s *Server) initLeafNodeSmapAndSendSubs(c *client) {
 	c.leaf.smap = make(map[string]int32)
 	for _, sub := range subs {
 		subj := string(sub.subject)
-		if c.isSpokeLeafNode() && !c.canSubscribe(subj) {
+		// Check perms regardless of role.
+		if !c.canSubscribe(subj) {
```

Comment:
I am concerned about this one since "blame" seem to indicate that there was a reason "isSpokeLeafNode()" was added: https://github.com/nats-io/nats-server/commit/6294c0b0c77279543d2fa7ac276613226fc20782 (meaning that prior to that commit there was no c.isSpokeLeafNode() check, only !c.canSubscribe(), so concerned that we could introduce a regression)

Reply:
I thought same thing around wider scoped subs being blocked but verified that they are not..

Reply:
On the git blame did we add a test?

Reply:
no, unfortunately.

Reply:
But it was linked to a user error report, so maybe worth going back and see if that causes the issue to reappear.

Reply:
I verified it works properly against that report.